include/execution.hpp

```diff
@@ -2171,7 +2172,257 @@ namespace std::execution {
+          , __data_(__data) {
+        }
+      };
+
+    enum class __state_t { __created, __started, __completed };
+
+    template <class _Sender>
+      struct __sh_state : __operation_base {
+        using __nullable_variant_t = __bind_front<__munique<__q<variant>>, tuple<>>;
+        using __error = __error_types_of_t<_Sender,
+                                           __empty_env,
+                                           __bind_front<__q<__variant>,
+                                                        exception_ptr>>;
+        using __value = __value_types_of_t<_Sender,
+                                           __empty_env,
+                                           __q<__decayed_tuple>,
+                                           __nullable_variant_t>;
+        using __receiver = __receiver<__value, __error>;
+
+        in_place_stop_source __stop_source_{};
+        vector<__operation_base*> __operation_states_;
+        connect_result_t<_Sender, __receiver> __op_state2_;
+        variant<__value, __error, set_stopped_t> __data_;
+
+        mutex __mutex_;
+        __state_t __state_{__state_t::__created};
+
+        explicit __sh_state(_Sender& __sndr)
+          : __op_state2_(
+                  connect((_Sender&&) __sndr,
+                          __receiver{*this, __stop_source_, __data_}))
+        {}
+
+        void __notify() noexcept override {
+          {
+            lock_guard __lock{__mutex_};
+            __state_ = __state_t::__completed;
+          }
+
+          for(auto __op_state: __operation_states_) {
+            __op_state->__notify();
+          }
+        }
+      };
+
+    template <class _SenderId, class _ReceiverId>
+      class __operation : public __operation_base {
+        using _Sender = __t<_SenderId>;
+        using _Receiver = __t<_ReceiverId>;
+
+        struct __on_stop_requested {
+          in_place_stop_source& __stop_source_;
+          void operator()() noexcept {
+            __stop_source_.request_stop();
+          }
+        };
+        using __on_stop = optional<typename stop_token_of_t<
+            env_of_t<_Receiver> &>::template callback_type<__on_stop_requested>>;
+
+        _Receiver __recvr_;
+        __on_stop __on_stop_{};
+        shared_ptr<__sh_state<_Sender>> __shared_state_;
+
+        template <class... _As>
+          requires tag_invocable<set_value_t, _Receiver, decay_t<_As>&...>
+        void __apply_values(tuple<_As...>& __tupl) noexcept {
+          try {
+            std::apply([&](auto&... __args) -> void {
+              execution::set_value((_Receiver&&) __recvr_, __args...);
+            }, __tupl);
+          } catch(...) {
+            execution::set_error((_Receiver&&) __recvr_, current_exception());
+          }
+        }
+
+        template <class... _As>
+          requires (!tag_invocable<set_value_t, _Receiver, decay_t<_As>&...>)
+        void __apply_values(tuple<_As...>&) noexcept {
+          assert(!"_Should never get here");
+        }
```

Comment:
Why do you need this overload?

Reply:
It's not required after redesigning storage to be a flat vector.